rdkit/Chem/SaltRemover.py

```diff
@@ -168,7 +170,7 @@ def _applyPattern(m, salt, notEverything):
     if dontRemoveEverything and len(Chem.GetMolFrags(mol)) <= 1:
       return mol
     modified = False
-    for i, salt in enumerate(self.salts):
+    for salt in self.salts:
       tMol = _applyPattern(mol, salt, dontRemoveEverything)
       if tMol is not mol:
```

Comment:
yeah, but the way it's written it doesn't actually check if the mol has been modified. It's testing whether or not it's the same object, which is only true here if the molecule has no atoms, `DeleteSubstructs()` somehow failed, or `notEverything` is true and the salt matched the whole molecule.
I think the intent of the original code:
```
    modified = False
    for i, salt in enumerate(self.salts):
      tMol = _applyPattern(mol, salt, dontRemoveEverything)
      if tMol is not mol:
        mol = tMol
        modified = True
        if dontRemoveEverything and len(Chem.GetMolFrags(mol)) <= 1:
          break
```
can be reproduced with:
```
    modified = False
    nats = mol.GetNumAtoms()
    for salt in self.salts:
      mol = _applyPattern(mol, salt, dontRemoveEverything)
      if mol.GetNumAtoms() != nats:
        nats = mol.GetNumAtoms()
        modified = True
        if dontRemoveEverything and len(Chem.GetMolFrags(mol)) <= 1:
          break
```